scripts/kconfig/kconfigfunctions.py

```diff
@@ -335,11 +335,47 @@ def dt_compat_enabled(kconf, _, compat):
     return "n"
 
 
+def dt_compat_get_string(kconf, _, compat, inst=0, ret_def='', prop='label'):
+    """
+    This function takes a 'compat' and an 'inst'ance to search for an EDT node that
+    matchs. If it finds an EDT node, it will look to see if that node has a name
+    property by the name of 'prop'. If the 'prop' exists it will return the string
+    representation of that 'prop' otherwise return ret_def.
+    """
+
+    ret_def_trimed = ret_def.strip()
+
+    try:
+        inst_no = int(inst)
+    except ValueError:
+        return ret_def_trimed
+
+    if doc_mode or edt is None:
+        return ret_def_trimed
+
+    for node in edt.nodes:
+        if compat not in node.instance_no:
+            continue
+
+        if node.instance_no[compat] != inst_no:
+            continue
+
+        if prop not in node.props:
+            break
+
+        if node.props[prop].val:
```

Comment:
I wonder if this check is needed. Maybe it could just be `return str(node.props[prop])`. If someone does e.g. `string-prop = ""`, then the empty string (which is falsy) is probably intentional.

Reply:
You are right! I did a lot of tests to understand the structure and missed this.